test/functional/p2p_sendtxrcncl.py

```diff
@@ -0,0 +1,191 @@
+        peer.wait_for_verack()
+        verack_index = [i for i, msg in enumerate(peer.messages) if msg.msgtype == b'verack'][0]
+        sendtxrcncl_index = [i for i, msg in enumerate(peer.messages) if msg.msgtype == b'sendtxrcncl'][0]
+        assert(sendtxrcncl_index < verack_index)
+        peer.peer_disconnect()
+
+        self.log.info('SENDTXRCNCL on pre-WTXID version should not be sent')
+        peer = self.nodes[0].add_p2p_connection(SendTxrcnclReceiver(), send_version=False, wait_for_verack=False)
+        pre_wtxid_version_msg = msg_version()
+        pre_wtxid_version_msg.nVersion = 70015
+        pre_wtxid_version_msg.strSubVer = P2P_SUBVERSION
+        pre_wtxid_version_msg.nServices = P2P_SERVICES
+        pre_wtxid_version_msg.relay = 1
+        peer.send_message(pre_wtxid_version_msg)
+        peer.wait_for_verack()
+        assert not peer.sendtxrcncl_msg_received
+        peer.peer_disconnect()
+
+        self.log.info('SENDTXRCNCL for fRelay=false should not be sent')
+        peer = self.nodes[0].add_p2p_connection(SendTxrcnclReceiver(), send_version=False, wait_for_verack=False)
+        no_txrelay_version_msg = msg_version()
+        no_txrelay_version_msg.nVersion = P2P_VERSION
+        no_txrelay_version_msg.strSubVer = P2P_SUBVERSION
+        no_txrelay_version_msg.nServices = P2P_SERVICES
+        no_txrelay_version_msg.relay = 0
+        peer.send_message(no_txrelay_version_msg)
+        peer.wait_for_verack()
+        assert not peer.sendtxrcncl_msg_received
+        peer.peer_disconnect()
+
+        self.log.info('valid SENDTXRCNCL received')
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(create_sendtxrcncl_msg())
+        self.wait_until(lambda : "sendtxrcncl" in self.nodes[0].getpeerinfo()[-1]["bytesrecv_per_msg"])
+        self.log.info('second SENDTXRCNCL triggers a disconnect')
+        peer.send_message(create_sendtxrcncl_msg())
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDTXRCNCL with initiator=responder=0 triggers a disconnect')
+        sendtxrcncl_no_role = create_sendtxrcncl_msg()
+        sendtxrcncl_no_role.initiator = False
+        sendtxrcncl_no_role.responder = False
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(sendtxrcncl_no_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDTXRCNCL with initiator=0 and responder=1 from inbound triggers a disconnect')
+        sendtxrcncl_wrong_role = create_sendtxrcncl_msg(initiator=False)
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(sendtxrcncl_wrong_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDTXRCNCL with version=0 triggers a disconnect')
```

Comment:
Could test the lawful case where version>=1 to ensure nodes deployed do not disconnect accidentally eventual reconciliation overhaul peers in the future.